tempesta_fw/ss_skb.c

```diff
@@ -1127,14 +1175,14 @@ ss_skb_unroll(SsSkbList *skb_list, struct sk_buff *skb)
 	if (unlikely(skb_cloned(skb)))
 		return ss_skb_unroll_slow(skb_list, skb);
 
-	/*
-	 * Note, that skb_gro_receive() drops reference to the
-	 * SKB's header via the __skb_header_release(). So, to
-	 * not break the things we must take reference back.
-	 */
 	ss_skb_queue_tail(skb_list, skb);
 	skb_walk_frags(skb, f_skb) {
 		if (f_skb->nohdr) {
+			/*
+			 * skb_gro_receive() drops reference to the SKB's header
+			 * via the __skb_header_release(). So, to not break the
+			 * things we must take reference back.
+			 */
```

Comment:
Frankly, I believe that original placement of this comment was better. This is a small piece of code, and it looks dense, nice, and very readable. The comment explains what happens in the logical block of code that follows it, and it's sufficient for understanding the block of code.  Placing multi-line comment in the middle of it just dilutes this small block of code and makes it a bit more difficult to follow.

I know that this may be a matter of taste. :-) I believe the code in the kernel in general is written more like to what I describe above.